# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/APZCTreeManagerChild.cpp
# Commit: 4230bdcf07ae
# Full Hash: 4230bdcf07ae69e03a9e241f10c201c38e6ca995
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2022-06-23 03:54:50
# Regressor Bug: 1773865
# File Overlap Count: 1
# Description:
#   Bug 1773865 - Dispatch an event on the window document when a pinch zoom gesture ends. r=botond,smaug,NeilDeakin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149283
# ==============================================================================

diff -r d1ddb1ce916b -r 4230bdcf07ae gfx/layers/ipc/APZCTreeManagerChild.cpp
--- a/gfx/layers/ipc/APZCTreeManagerChild.cpp	Wed Jun 22 16:08:22 2022 +0000
+++ b/gfx/layers/ipc/APZCTreeManagerChild.cpp	Wed Jun 22 16:15:57 2022 +0000
@@ -201,5 +201,19 @@
   return IPC_OK();
 }
 
+mozilla::ipc::IPCResult APZCTreeManagerChild::RecvNotifyScaleGestureComplete(
+    const ScrollableLayerGuid::ViewID& aScrollId, float aScale) {
+  // This will only get sent from the GPU process to the parent process, so
+  // this function should never get called in the content process.
+  MOZ_ASSERT(XRE_IsParentProcess());
+  MOZ_ASSERT(NS_IsMainThread());
+
+  if (mCompositorSession && mCompositorSession->GetWidget()) {
+    APZCCallbackHelper::NotifyScaleGestureComplete(
+        mCompositorSession->GetWidget(), aScale);
+  }
+  return IPC_OK();
+}
+
 }  // namespace layers
 }  // namespace mozilla